etl/steps/data/garden/energy/2022-12-13/uk_historical_electricity.py

```diff
@@ -0,0 +1,213 @@
+        "total_net_imports__twh": "net_imports",
+    }
+
+    # Select necessary columns from electricity mix dataset.
+    df_elec = df_elec[list(elec_columns)].rename(columns=elec_columns)
+
+    # Select UK data from Ember dataset.
+    df_elec = df_elec[df_elec["country"] == "United Kingdom"].reset_index(drop=True)
+
+    return df_elec
+
+
+def prepare_beis_data(df_beis: pd.DataFrame) -> pd.DataFrame:
+    """Select (and rename) columns from the UK historical electricity data from BEIS.
+
+    Parameters
+    ----------
+    df_beis : pd.DataFrame
+        Combined data for UK historical electricity data from BEIS.
+
+    Returns
+    -------
+    df_beis : pd.DataFrame
+        Selected columns from the UK historical electricity data.
+
+    """
+    df_beis = df_beis.copy()
+
+    # Select columns and rename them conveniently.
+    beis_columns = {
+        "country": "country",
+        "year": "year",
+        "coal": "coal_generation",
+        "oil": "oil_generation",
+        "electricity_generation": "total_generation",
+        "gas": "gas_generation",
+        "hydro": "hydro_generation",
+        "nuclear": "nuclear_generation",
+        "net_imports": "net_imports",
+        "implied_efficiency": "implied_efficiency",
+        "wind_and_solar": "wind_and_solar_generation",
+    }
+    df_beis = df_beis[list(beis_columns)].rename(columns=beis_columns)
+
+    return df_beis
+
+
+def combine_beis_and_electricity_mix_data(df_beis: pd.DataFrame, df_elec: pd.DataFrame) -> pd.DataFrame:
+    """Combine BEIS data on UK historical electricity with the electricity mix data (after having selected rows for only
+    the UK).
+
+    There are different processing steps done to the data, see comments below in the code.
+
+    Parameters
+    ----------
+    df_beis : pd.DataFrame
+        Selected data from BEIS on UK historical electricity.
+    df_elec : pd.DataFrame
+        Selected data from the electricity mix (after having selected rows for the UK).
+
+    Returns
+    -------
+    df_combined : pd.DataFrame
+        Combined and processed data with a verified index.
+
+    """
+    # In the BEIS dataset, wind and solar are given as one joined variable.
+    # Check if we can ignore it (since it's better to have the two sources separately).
+    # Find the earliest year informed in the electricity mix for solar or wind generation.
+    solar_or_wind_first_year = df_elec[df_elec["wind_generation"].notnull() | df_elec["solar_generation"].notnull()][
+        "year"
+    ].min()
+    # Now check that, prior to that year, all generation from solar and wind was zero.
+    assert df_beis[df_beis["year"] < solar_or_wind_first_year]["wind_and_solar_generation"].fillna(0).max() == 0
+    # Therefore, since wind and solar is always zero (prior to the beginning of the electricity mix data)
+    # we can ignore this column from the BEIS dataset.
+    df_beis = df_beis.drop(columns="wind_and_solar_generation")
```

Comment:
For consistency with other uses that you make of `pd.DataFrame.drop`, I'd provide the value for argument `columns` as a list, i.e.

```python
df_beis = df_beis.drop(columns=["wind_and_solar_generation"])
```

Reply:
I've changed all instances of drop(columns=...) to ingest lists.